azure-keyvault/azure/keyvault/keys/_client.py

```diff
@@ -58,236 +44,552 @@ def create_config(**kwargs):
         config.retry = RetryPolicy(**kwargs)
         config.redirect = RedirectPolicy(**kwargs)
 
-        # TODO: these are requests-specific
-        config.cert = config.timeout = None
-        config.verify = True
         return config
 
-    def __init__(self, vault_url, credentials, config=None, transport=None):
-        # type: (str, Any, Configuration, HttpTransport) -> None
+    def __init__(self, vault_url, credentials, config=None, transport=None, **kwargs):
+        # type: (str, Any, Configuration, HttpTransport, Mapping[str, Any]) -> None
+        # TODO: update type hint for credentials
+        if not credentials:
+            raise ValueError("credentials")
+
+        if not vault_url:
+            raise ValueError("vault_url")
+
         self.vault_url = vault_url.strip("/")
-        config = config or KeyClient.create_config()
-        transport = RequestsTransport(config)
+        config = config or KeyClient.create_config(**kwargs)
+        transport = RequestsTransport(config.connection)
```

Comment:
>transport = RequestsTransport(config.connection) [](start = 8, length = 48)

Shouldn't this only create transport if one wasn't specified?

Reply:
Yes. This is leftover from when `azure-core` was incomplete. We've left the constructor alone because it has to be redone anyway before it can wrap generated code, which is a next step. My hope for this PR is to solidify the client surface and a set of tests.